packages/tools/examples/contourApi/index.ts

```diff
@@ -0,0 +1,311 @@
+element2.style.width = size;
+element2.style.height = size;
+
+// Disable right click context menu so we can have right click tool
+element1.oncontextmenu = (e) => e.preventDefault();
+element2.oncontextmenu = (e) => e.preventDefault();
+
+viewportGrid.appendChild(element1);
+viewportGrid.appendChild(element2);
+
+content.appendChild(viewportGrid);
+
+createInfoSection(content, { title: 'Drawing' })
+  .addInstruction('Left click and drag to draw a contour')
+  .openNestedSection()
+  .addInstruction(
+    'If you join the contour together it will be closed, otherwise releasing the mouse will create an open contour (freehand line)'
+  );
+
+createInfoSection(content, { title: 'Editing' })
+  .addInstruction(
+    'Left click and drag on the line of an existing contour to edit it'
+  )
+  .openNestedSection()
+  .addInstruction('Closed Contours')
+  .openNestedSection()
+  .addInstruction(
+    'Drag the line and a preview of the edit will be displayed. Release the mouse to complete the edit. You can cross the original contour multiple times in one drag to do a complicated edit in one movement.'
+  )
+  .closeNestedSection();
+
+createInfoSection(content, { title: 'Contour Utilities' })
+  .addInstruction(
+    'Find Contour Holes: Analyzes contours to detect holes within them. Results are logged to console.'
+  )
+  .addInstruction(
+    'Smooth Polylines: Smooth contours by choosing control points in the contour and applying a spline interpolation on them. Note: it alters the contour shape if there is few points'
+  )
+  .addInstruction(
+    'Remove Small Islands: Filters out closed contours smaller than area threshold (3 cm2 units). Only affects closed contours, preserves open polylines.'
+  )
+  .addInstruction(
+    'Decimate Polylines: Simplifies polylines by removing points using Ramer-Douglas-Peucker algorithm with epsilon tolerance of 2.0 units. Shows reduction percentage.'
+  );
+
+addSegmentIndexDropdown();
+addButtonToToolbar({
+  title: 'Remove Contour Holes',
+  onClick: () => {
+    const segmentIndex =
+      segmentation.segmentIndex.getActiveSegmentIndex(segmentationId);
+    segmentation.utilities.removeContourHoles(segmentationId, segmentIndex);
+    const renderingEngine = getRenderingEngine(renderingEngineId);
+    renderingEngine.render();
+  },
+});
+
+addButtonToToolbar({
+  title: 'Remove Small Islands',
+  onClick: () => {
+    const segmentIndex =
+      segmentation.segmentIndex.getActiveSegmentIndex(segmentationId);
+    segmentation.utilities.removeContourIslands(segmentationId, segmentIndex, {
+      threshold: 10,
+    });
+    const renderingEngine = getRenderingEngine(renderingEngineId);
+    renderingEngine.render();
+  },
+});
+
+addButtonToToolbar({
+  title: 'Smooth Polylines',
+  onClick: () => {
+    const segmentIndex =
+      segmentation.segmentIndex.getActiveSegmentIndex(segmentationId);
+    segmentation.utilities.smoothContours(segmentationId, segmentIndex, {
+      knotsRatioPercentage: 60,
```

Comment:
can we just call it threshold? or something general? not everyone knows what is knotsRatioPercentage